src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,137 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private static final int INITIAL_SIZE = 0;
+    private static final int INITIAL_INDEX = 0;
+    private static final String TEXT_OF_EXCEPTION = "Invalid index outside"
+            + " the dimension of the array";
+    private int size;
+    private T[] elementData;
+
+    public ArrayList() {
+        size = INITIAL_SIZE;
+        elementData = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            increaseElementDataLength();
+        }
+        addValueToElementData(value);
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < INITIAL_INDEX) {
+            throw new ArrayListIndexOutOfBoundsException(TEXT_OF_EXCEPTION);
+        } else if (index == elementData.length - 1) {
+            increaseElementDataLength();
+            addValueToElementData(value);
+        } else {
+            rightShiftOfElements(value, index);
+        }
```

Comment:
`else` clause is redundat because of exiting from the method when throwing exception in 31 line.

```suggestion
        if (index == elementData.length - 1) {
            increaseElementDataLength();
        } 
        System.arraycopy(data, index, data, index + 1, size - index);
        data[index] = value;
        size++;
        }
```